src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -304,140 +297,233 @@ namespace nbl::ext::imgui
 			params.subresourceRange = regions.subresource;
 			params.image = core::smart_refctd_ptr(image);
 
-			m_fontTexture = m_device->createImageView(std::move(params));
+			m_fontAtlasTexture = m_device->createImageView(std::move(params));
 		}
 		
         ISemaphore::future_t<IQueue::RESULT> retval(IQueue::RESULT::SUCCESS);
         retval.set({sInfo.scratchSemaphore.semaphore,sInfo.scratchSemaphore.value});
         return retval;
 	}
 
-	void prepareKeyMapForDesktop()
-	{
-		ImGuiIO& io = ImGui::GetIO();
-		// TODO:
-		// Keyboard mapping. ImGui will use those indices to peek into the io.KeysDown[] array.
-		//io.KeyMap[ImGuiKey_Tab] = MSDL::SDL_SCANCODE_TAB;
-		//io.KeyMap[ImGuiKey_LeftArrow] = MSDL::SDL_SCANCODE_LEFT;
-		//io.KeyMap[ImGuiKey_RightArrow] = MSDL::SDL_SCANCODE_RIGHT;
-		//io.KeyMap[ImGuiKey_UpArrow] = MSDL::SDL_SCANCODE_UP;
-		//io.KeyMap[ImGuiKey_DownArrow] = MSDL::SDL_SCANCODE_DOWN;
-		//io.KeyMap[ImGuiKey_PageUp] = MSDL::SDL_SCANCODE_PAGEUP;
-		//io.KeyMap[ImGuiKey_PageDown] = MSDL::SDL_SCANCODE_PAGEDOWN;
-		//io.KeyMap[ImGuiKey_Home] = MSDL::SDL_SCANCODE_HOME;
-		//io.KeyMap[ImGuiKey_End] = MSDL::SDL_SCANCODE_END;
-		//io.KeyMap[ImGuiKey_Insert] = MSDL::SDL_SCANCODE_INSERT;
-		//io.KeyMap[ImGuiKey_Delete] = MSDL::SDL_SCANCODE_DELETE;
-		//io.KeyMap[ImGuiKey_Backspace] = MSDL::SDL_SCANCODE_BACKSPACE;
-		//io.KeyMap[ImGuiKey_Space] = MSDL::SDL_SCANCODE_SPACE;
-		//io.KeyMap[ImGuiKey_Enter] = MSDL::SDL_SCANCODE_RETURN;
-		//io.KeyMap[ImGuiKey_Escape] = MSDL::SDL_SCANCODE_ESCAPE;
-		//io.KeyMap[ImGuiKey_KeyPadEnter] = MSDL::SDL_SCANCODE_KP_ENTER;
-		//io.KeyMap[ImGuiKey_A] = MSDL::SDL_SCANCODE_A;
-		//io.KeyMap[ImGuiKey_C] = MSDL::SDL_SCANCODE_C;
-		//io.KeyMap[ImGuiKey_V] = MSDL::SDL_SCANCODE_V;
-		//io.KeyMap[ImGuiKey_X] = MSDL::SDL_SCANCODE_X;
-		//io.KeyMap[ImGuiKey_Y] = MSDL::SDL_SCANCODE_Y;
-		//io.KeyMap[ImGuiKey_Z] = MSDL::SDL_SCANCODE_Z;
-	}
-
 	static void adjustGlobalFontScale()
 	{
 		ImGuiIO& io = ImGui::GetIO();
 		io.FontGlobalScale = 1.0f;
 	}
 
-	void UI::UpdateDescriptorSets()
+	void UI::handleMouseEvents(const core::SRange<const nbl::ui::SMouseEvent>& events, const ui::IWindow* window) const
 	{
-		IGPUDescriptorSet::SDescriptorInfo info;
+		auto& io = ImGui::GetIO();
+
+		const auto cursorPosition = window->getCursorControl()->getPosition();
+		const auto mousePixelPosition = nbl::hlsl::float32_t2(cursorPosition.x, cursorPosition.y) - nbl::hlsl::float32_t2(window->getX(), window->getY());
```

Comment:
have the user do all this translation/filtering stuff outside, never ask for an `IWindow` for IMGui

P.S. also user should filter out any clicks outside of ImGUIs region of interest

Reply:
resolved